### ISSUE TITLE
stack deploy does not use env-vars from shell

### ISSUE THREAD

**Issue by pdonorio** [docker/docker#31319]
**Description**

Trying to use my docker-compose file to stack my swarm.
I am testing here and there features and environment variables are a must.
But, I can't seem to forward them from my shell to the swarm as explained by the documentation.

Note: I am on Docker for Mac Beta, but never had an issue on anything. 
I don't have the time this week to switch to stable to make sure it does not depend on the beta, but if necessary I can double check the very next one.

**Steps to reproduce the issue:**

1. init the swarm, with one node and one manager
2. I set my variable in my bash:
```bash
bash-4.4$ echo $MYNAME
something
```
3. Create and deploy the stack with a simple compose:
```
bash-4.4$ cat docker-compose.yml
version: '3'

services:
  client:
    image: postgres:9.6
    command: sleep 1234567890
    environment:
      - MYNAME
    networks:
      - mynet

  server:
    image: postgres:9.6
    networks:
      mynet:
        aliases:
          - db

networks:
  mynet:

bash-4.4$ docker deploy -c docker-compose.yml test
```

**Describe the results you received:**

```
docker service inspect test_client | jq .[0].Spec.TaskTemplate.ContainerSpec.Env
[
  "MYNAME="
]
```

**Describe the results you expected:**

```
docker service inspect test_client | jq .[0].Spec.TaskTemplate.ContainerSpec.Env
[
  "MYNAME=something"
]
```

**Additional information you deem important (e.g. issue happens only occasionally):**

It works as expected if I do `docker-compose up client`.

**Output of `docker version`:**

```
Client:
 Version:      17.03.0-ce-rc1
 API version:  1.26
 Go version:   go1.7.5
 Git commit:   ce07fb6
 Built:        Mon Feb 20 10:12:38 2017
 OS/Arch:      darwin/amd64

Server:
 Version:      17.03.0-ce-rc1
 API version:  1.26 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   ce07fb6
 Built:        Mon Feb 20 10:12:38 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

**Output of `docker info`:**

```
Containers: 7
 Running: 2
 Paused: 0
 Stopped: 5
Images: 85
Server Version: 17.03.0-ce-rc1
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
Swarm: active
 NodeID: e775z7lt77l793fipjsaem4bv
 Is Manager: true
 ClusterID: v910js09mbrhxc8o1804deb09
 Managers: 1
 Nodes: 1
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 192.168.65.2
 Manager Addresses:
  192.168.65.2:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: aa8187dbd3b7ad67d8e5e3a15115d3eef43a7ed1
runc version: 9df8b306d01f59d3a8029be411de015b7304dd8f
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.11-moby
Operating System: Alpine Linux v3.5
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 1.952 GiB
Name: moby
ID: BIWR:ZM3N:3P6O:JFA4:KTZJ:EHDG:7UBY:5TPV:CK6K:I5BI:2O6M:OMWL
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 58
 Goroutines: 177
 System Time: 2017-02-24T08:31:08.755419506Z
 EventsListeners: 3
Username: pdonorio
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

On Mac OS Sierra 10.12.3 (16D32),

Docker for Mac Version 17.03.0-ce-rc1-mac2 (15497)
Channel: beta 7c93573425

**Comment by thaJeztah** [comment]
ping @dnephin @vdemeester

**Comment by dnephin** [comment]
#30781 was opened a while ago to fix this

**Comment by vdemeester** [comment]
oh boy  didn't see it, I need some sleep I guess 😴

**Comment by pdonorio** [comment]
Sorry to bother: if I got it right there is already a PR for this which would work. 
May I help to get this further?

**Comment by vdemeester** [comment]
@pdonorio yes, and the PR has been merged 👼 This should be available for 17.04 😉

**Comment by pdonorio** [comment]
Thank you!
